main/src/ca/uwaterloo/flix/language/phase/unification/CofiniteIntSet.scala

```diff
@@ -0,0 +1,137 @@
+/*
+ * Copyright 2024 Jonathan Lindegaard Starup
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package ca.uwaterloo.flix.language.phase.unification
+
+import scala.collection.immutable.SortedSet
+
+/**
+  * Represents a finite or co-finite set with an infinite universe of integers.
+  *
+  * All sets are either a [[SortedSet]] or a complement of it.
+  *
+  * No finite set is ever equivalent to universe.
+  */
+sealed trait CofiniteIntSet {
+
+  import CofiniteIntSet.{Compl, Set}
+
+  /** Returns `true` if `this` is [[CofiniteIntSet.universe]]. */
```

Comment:
Why are all your comments on 1 line? They should be formatted on multiple lines :)

Reply:
Yeah, but we don't do that anywhere else...

Reply:
WIll Intellij not screw up the formatting when I open the file?

Reply:
No intellij also likes it